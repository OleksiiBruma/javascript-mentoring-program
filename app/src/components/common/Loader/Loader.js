import React from "react";
import { LinearProgress } from "@material-ui/core";

export const Loader = (props) => {
    const completed = 100;
  return props.isLoading ? <LinearProgress /> : <LinearProgress variant="determinate" value={completed}/>;
};
