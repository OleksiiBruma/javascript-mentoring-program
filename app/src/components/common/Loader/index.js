import { Loader } from "./Loader";
import { connect } from "react-redux";

export default connect(state => ({ isLoading: state.loader }))(Loader);
