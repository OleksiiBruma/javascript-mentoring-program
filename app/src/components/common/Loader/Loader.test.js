import renderer from "react-test-renderer";
import React from "react";
import { Loader } from "../Loader/Loader";

it("Loader return null when isLoading = false", () => {
  const component = renderer.create(<Loader isLoading={false} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
it("return Loader", () => {
  const component = renderer.create(<Loader isLoading={true} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
