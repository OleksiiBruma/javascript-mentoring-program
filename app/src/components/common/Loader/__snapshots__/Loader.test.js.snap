// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader return null when isLoading = false 1`] = `
<div
  aria-valuenow={100}
  className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
  role="progressbar"
>
  <div
    className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
    style={
      Object {
        "transform": "translateX(0%)",
      }
    }
  />
</div>
`;

exports[`return Loader 1`] = `
<div
  className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
  role="progressbar"
>
  <div
    className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
    style={Object {}}
  />
  <div
    className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
    style={Object {}}
  />
</div>
`;
